lib/Frontend/Frontend.cpp

```diff
@@ -165,6 +165,19 @@ SerializationOptions CompilerInvocation::computeSerializationOptions(
     serializationOpts.ImportedHeader = opts.ImplicitObjCHeaderPath;
   serializationOpts.ModuleLinkName = opts.ModuleLinkName;
   serializationOpts.ExtraClangOptions = getClangImporterOptions().ExtraArgs;
+  
+  if (opts.EmitSymbolGraph) {
+    if (!opts.SymbolGraphOutputDir.empty()) {
+      serializationOpts.SymbolGraphOutputDir = opts.SymbolGraphOutputDir;
+    } else {
+      serializationOpts.SymbolGraphOutputDir = serializationOpts.OutputPath;
+    }
+    SmallString<256> OutputDir(serializationOpts.SymbolGraphOutputDir);
+    llvm::sys::fs::make_absolute(OutputDir);
+    llvm::sys::path::remove_filename(OutputDir);
+    serializationOpts.SymbolGraphOutputDir = OutputDir.str().str();
```

Comment:
Is there a specific reason `.str()` is called twice here?

Reply:
The first creates a `StringRef`, and the second copies into a `std::string`. It's just a naming coincidence.

Reply:
Interesting, thanks for the explanation!